core/dbt/parser/schemas.py

```diff
@@ -360,37 +362,10 @@ def render_test_update(self, node, config, builder):
         node.depends_on.add_macro(macro_unique_id)
         if (macro_unique_id in
                 ['macro.dbt.test_not_null', 'macro.dbt.test_unique']):
-            self.update_parsed_node(node, config)
-            # manually set configs
-            # note: this does not respect generate_alias_name() macro
-            if builder.alias is not None:
-                node.unrendered_config['alias'] = builder.alias
-                node.config['alias'] = builder.alias
-                node.alias = builder.alias
-            if builder.severity is not None:
-                node.unrendered_config['severity'] = builder.severity
-                node.config['severity'] = builder.severity
-            if builder.enabled is not None:
-                node.unrendered_config['enabled'] = builder.enabled
-                node.config['enabled'] = builder.enabled
-            if builder.where is not None:
-                node.unrendered_config['where'] = builder.where
-                node.config['where'] = builder.where
-            if builder.limit is not None:
-                node.unrendered_config['limit'] = builder.limit
-                node.config['limit'] = builder.limit
-            if builder.warn_if is not None:
-                node.unrendered_config['warn_if'] = builder.warn_if
-                node.config['warn_if'] = builder.warn_if
-            if builder.error_if is not None:
-                node.unrendered_config['error_if'] = builder.error_if
-                node.config['error_if'] = builder.error_if
-            if builder.fail_calc is not None:
-                node.unrendered_config['fail_calc'] = builder.fail_calc
-                node.config['fail_calc'] = builder.fail_calc
-            if builder.store_failures is not None:
-                node.unrendered_config['store_failures'] = builder.store_failures
-                node.config['store_failures'] = builder.store_failures
+            config_call_dict = builder.get_static_config()
```

Comment:
As far as I can tell from reading the code, the unrendered_config only has actual unrendered_config from dbt_project. The config from SQL files is rendered by jinja in the rendering process. So we probably do need to put config in schema files through a get_rendered step, which this pr is not doing yet. We *could* save the schema file config dictionary in rendered and unrendered form, but I think we'd have to statically parse config in SQL files to get unrendered config from there.

Reply:
And we could save the unrendered config from tests too, but it would require refactoring that update_parsed_node(_config) codepath, because it doesn't have a source of unrendered_config for nodes.

Reply:
Note: I think the experimental parser doesn't render configs either.

Reply:
> So we probably do need to put config in schema files through a get_rendered step, which this pr is not doing yet.

Ah. I think we'll want them to have the same rendering context as other preexisting `schema.yml` properties (e.g. `meta`, `tags`).

> We could save the schema file config dictionary in rendered and unrendered form

Yes, this is what we do for config from `dbt_project.yml`. It's not a must-have right now, just a way to keep making `state:modified` comparisons better and better.

> I think we'd have to statically parse config in SQL files to get unrendered config from there... I think the experimental parser doesn't render configs either.

Yup, I think this will come from making our tree-sitter grammar able to parse unrendered configs, detect Jinja expressions, and then pass those expressions in for rendering. This is what I'd love to work toward, ultimately: parse/extract _all_ unrendered configs, then render all configs in one big Jinja-rendering step.

Reply:
Noted that long con here: https://github.com/dbt-labs/dbt/issues/3680